src/orange/universes/SimpleUnitTracker.hh

```diff
@@ -214,6 +218,37 @@ SimpleUnitTracker::intersect(const LocalState& state, real_type max_dist) const
     return result;
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Calculate nearest distance to a surface in any direction.
+ *
+ * The safety calculation uses a very limited method for calculating the safety
+ * distance: it's the nearest distance to any surface, for a certain subset of
+ * surfaces.  Other surface types will return a safety distance of zero.
+ * Complex surfaces might return the distance to internal surfaces that do not
+ * represent the edge of a cell. Such distances are conservative but will
+ * necessarily slow down the simulation.
+ */
+CELER_FUNCTION real_type SimpleUnitTracker::safety(const Real3& pos,
+                                                   VolumeId     volid) const
+{
+    CELER_EXPECT(volid);
+
+    real_type result = numeric_limits<real_type>::infinity();
+
+    // Calculate minimim distance to all local faces
+    VolumeView vol{params_.volumes, volid};
+    auto       calc_safety = make_surface_action(Surfaces{params_.surfaces},
+                                           detail::CalcSafetyDistance{pos});
+    for (SurfaceId surface : vol.faces())
+    {
+        result = celeritas::min(result, calc_safety(surface));
```

Comment:
How "common" will the "is not simple_safety" case?  If it is "not" uncommon, would it make sense to records with the list of volumes that one of more of those volumes is not a "simple_safety" and thus the loop is necessary (The safety will be always be zero if at least one of the volume does not have a simple safety - which is known a geometry construction time).

Reply:
Since ORANGE in Celeritas is currently just a "test" geometry it's pretty simple indeed... but with the "flags" that each volume has it would be very easy to add this feature. Good idea!